src/frontend/src/common/queries/trees.ts

```diff
@@ -200,3 +202,58 @@ export function useDeleteTree({
     },
   });
 }
+
+/**
+ * edit trees
+ */
+
+interface EditTreePayloadType {
+  name: string;
+}
+
+interface TreeEditRequestType {
+  treeIdToEdit: string;
+  newTreeName: string;
+}
+
+interface TreeEditResponseType {
+  id: string;
+}
+
+type TreeEditCallbacks = MutationCallbacks<TreeEditResponseType>;
+
+export async function editTree({
+  treeIdToEdit,
+  newTreeName,
+}: TreeEditRequestType): Promise<TreeEditResponseType> {
+  const payload: EditTreePayloadType = {
+    name: newTreeName,
+  };
+  const response = await fetch(API_URL + API.PHYLO_TREES_V2 + treeIdToEdit, {
+    ...DEFAULT_PUT_OPTIONS,
+    ...DEFAULT_HEADERS_MUTATION_OPTIONS,
```

Comment:
I strongly believe we should fold `DEFAULT_HEADERS_MUTATION_OPTIONS` into `DEFAULT_PUT_OPTIONS` (this is currently done with `DEFAULT_POST_OPTIONS`). The only prior existing use of `DEFAULT_PUT_OPTIONS` in the codebase already does the same `...DEFAULT_HEADERS_MUTATION_OPTIONS` spread as is being done here (this happens with `updateUserInfo` in auth.ts). I just can't really imagine us sending a PUT request that isn't ultimately sending JSON to the BE unless it's something so special and unusual that it can get a special case to handle its weirdo request headers.